test/e2e/framework/node/helper.go

```diff
@@ -175,3 +179,78 @@ func IsARM64(node *v1.Node) bool {
 
 	return false
 }
+
+// patchNode sends a patch request to update the Node.
+func patchNode(ctx context.Context, client clientset.Interface, old *v1.Node, new *v1.Node) error {
+	oldData, err := json.Marshal(old)
+	if err != nil {
+		return err
+	}
+
+	newData, err := json.Marshal(new)
+	if err != nil {
+		return err
+	}
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, &v1.Node{})
+	if err != nil {
+		return fmt.Errorf("failed to create merge patch for node %q: %w", old.Name, err)
+	}
+	_, err = client.CoreV1().Nodes().Patch(ctx, old.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{}, "status")
+	return err
+}
+
+// AddExtendedResource adds a fake resource to the Node.
+func AddExtendedResource(ctx context.Context, clientSet clientset.Interface, nodeName string, extendedResourceName v1.ResourceName, extendedResourceQuantity resource.Quantity) {
+	extendedResource := v1.ResourceName(extendedResourceName)
+
+	ginkgo.By("Adding a custom resource")
+	OriginalNode, err := clientSet.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	node := OriginalNode.DeepCopy()
+	node.Status.Capacity[extendedResource] = extendedResourceQuantity
+	node.Status.Allocatable[extendedResource] = extendedResourceQuantity
```

Comment:
Can you just construct the patch directly? Sort of like this: https://github.com/kubernetes/kubernetes/blob/d9187112c2b2f723878947e666387d3640728e0d/test/e2e/scheduling/limit_range.go#L315-L328

```go
extendedResourceList := v1.ResourceList{
	extendedResource: extendedResourceQuantity
}
patchPayload := json.Marshal(v1.Node{
	Status: v1.NodeStatus{
		Capacity: extendedResourceList,
		Allocatable: extendedResourceList,
	},
})
```

Reply:
Done.